From e77ebb2c340685b24a786164fb152779d5e57d6c Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 17:47:49 +0530
Subject: [PATCH 070/500] OpenWrt:
 702-net-ethernet-mtk_eth_soc-enable-threaded-NAPI.patch

net: ethernet: mtk_eth_soc: enable threaded NAPI

This can improve performance under load by ensuring that NAPI processing is
not pinned on CPU 0.

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: Iab1676792a5c4b371416244370ed962abf4aaaec
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 9409f9ef4f80fc5132bb65663a6f76032da3306a)
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
index 20afe79f380a..7f3a8b77f9ae 100644
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -4978,6 +4978,8 @@ static int mtk_probe(struct platform_device *pdev)
 	 * for NAPI to work
 	 */
 	init_dummy_netdev(&eth->dummy_dev);
+	eth->dummy_dev.threaded = 1;
+	strcpy(eth->dummy_dev.name, "mtk_eth");
 	netif_napi_add(&eth->dummy_dev, &eth->tx_napi, mtk_napi_tx);
 	netif_napi_add(&eth->dummy_dev, &eth->rx_napi, mtk_napi_rx);
 
-- 
2.34.1

